playground/TestShop/AppHost/ParameterExtensions.cs

```diff
@@ -0,0 +1,117 @@
+    /// Creates a password parameter that in the development environment is generated once and stored in the user secrets store.
+    /// </summary>
+    /// <remarks>
+    /// The password is only stable when in the development environment and the application is running. In all other cases, the password is generated each time.
+    /// </remarks>
+    public static IResourceBuilder<ParameterResource> CreateStablePassword(this IDistributedApplicationBuilder builder, string name,
+        bool lower = true, bool upper = true, bool numeric = true, bool special = true,
+        int minLower = 0, int minUpper = 0, int minNumeric = 0, int minSpecial = 0)
+    {
+        ParameterDefault generatedPassword = new GenerateParameterDefault
+        {
+            MinLength = 22, // enough to give 128 bits of entropy when using the default 67 possible characters. See remarks in PasswordGenerator.Generate
+            Lower = lower,
+            Upper = upper,
+            Numeric = numeric,
+            Special = special,
+            MinLower = minLower,
+            MinUpper = minUpper,
+            MinNumeric = minNumeric,
+            MinSpecial = minSpecial
+        };
+
+        if (builder.Environment.IsDevelopment() && builder.ExecutionContext.IsRunMode)
+        {
+            // In development mode, generate a new password each time the application starts
+            generatedPassword = new UserSecretsParameterDefault(builder.Environment.ApplicationName, name, generatedPassword);
+        }
+
+        var parameterResource = new ParameterResource(name, parameterDefault => GetParameterValue(builder.Configuration, name, parameterDefault), true)
+        {
+            Default = generatedPassword
+        };
+
+        return ResourceBuilder.Create(parameterResource, builder);
+    }
+
+    private static string GetParameterValue(IConfiguration configuration, string name, ParameterDefault? parameterDefault)
+    {
+        var configurationKey = $"Parameters:{name}";
+        return configuration[configurationKey]
+            ?? parameterDefault?.GetDefaultValue()
+            ?? throw new DistributedApplicationException($"Parameter resource could not be used because configuration key '{configurationKey}' is missing and the Parameter has no default value."); ;
+    }
+
+    sealed class UserSecretsParameterDefault(string applicationName, string parameterName, ParameterDefault parameterDefault) : ParameterDefault
+    {
+        public override string GetDefaultValue()
+        {
+            var value = parameterDefault.GetDefaultValue();
+            var configurationKey = $"Parameters:{parameterName}";
+            TrySetUserSecret(applicationName, configurationKey, value);
+            return value;
+        }
+
+        public override void WriteToManifest(ManifestPublishingContext context) => parameterDefault.WriteToManifest(context);
+
+        private static bool TrySetUserSecret(string applicationName, string name, string value)
```

Comment:
```suggestion
        private static void SetUserSecret(string applicationName, string name, string value)
```

Nothing uses this bool.